### PR TITLE
Fix naming of configuration parameters, add support for keycloak

### PR DESCRIPTION
**Pull request by lionick**

### CODE REVIEW

File: lib/Auth/Process/OIDCAuthorize.php [hunk 1]
```diff
@@ -130,13 +144,21 @@ public function process(&$request)
...
-    if (empty($client_id)) {
-        throw new Error\Error(
-            ['UNHANDLEDEXCEPTION', 'Request missing saml:RelayState']
-        );
-    }
+    $client_id = null;
 
+    if (!empty($request['saml:RelayState']) 
+      && !empty($this->keycloakSp) 
+      && strpos($request['Destination']['entityid'],
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/OIDCAuthorize.php]: Exact match. Test with https://{{ rciam_hostname }}/auth/realms/egi

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/OIDCAuthorize.php [hunk 2]
```diff
@@ -130,13 +144,21 @@ public function process(&$request)
...
-            ['UNHANDLEDEXCEPTION', 'Request missing saml:RelayState']
-        );
-    }
+    $client_id = null;
 
+    if (!empty($request['saml:RelayState']) 
+      && !empty($this->keycloakSp) 
+      && $request['Destination']['entityid'] == 
+                $this->keycloakSp) {
+      $client_id = explode('.', $request['saml:RelayState'], 3)[2];
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/OIDCAuthorize.php]: We should log error "[OIDCAuthorize] Client ID is empty" and throw an UNHANDLEDEXCEPTION 'Oops something went wrong' error if the client_id is empty

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/OIDCAuthorize.php (hunk 2 above)
Comment (lionick) [reply lib/Auth/Process/OIDCAuthorize.php]: I have just added an UNHANDLEDEXCEPTION with a meaningful message as this is only printed at logs. User sees a screen with the tracking number of the error.

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**